core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -176,7 +176,10 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
     packages = Option(packages).orElse(sparkProperties.get("spark.jars.packages")).orNull
     packagesExclusions = Option(packagesExclusions)
       .orElse(sparkProperties.get("spark.jars.excludes")).orNull
-    deployMode = Option(deployMode).orElse(env.get("DEPLOY_MODE")).orNull
+    deployMode = Option(deployMode)
+      .orElse(sparkProperties.get("spark.deployMode"))
```

Comment:
why not just reuse `spark.submit.deployMode`? Would that work? Now there's both of these and it's a little confusing.